src/Struct/FindologicProduct.php

```diff
@@ -338,10 +337,18 @@ protected function getTranslatedSeoPath(): ?string
     {
         $salesChannel = $this->salesChannelContext->getSalesChannel();
         $seoUrlCollection = $this->product->getSeoUrls()->filterBySalesChannelId($salesChannel->getId());
+        $collectionWithoutDeletedEntities = $this->filterDeletedSeoUrls($seoUrlCollection);
 
-        /** @var SeoUrlEntity|null $seoUrlEntity */
-        $seoUrlEntity = $this->getTranslatedEntity($seoUrlCollection);
+        $seoUrlEntities = $this->getTranslatedEntities($collectionWithoutDeletedEntities);
+        if (!$seoUrlEntities) {
+            return null;
+        }
+
+        $canonicalSeoUrl = $seoUrlEntities->filter(function (SeoUrlEntity $entity) {
+            return $entity->getIsCanonical();
+        })->first();
```

Comment:
Is `->first()` handling zero results after filter?

Reply:
`first()` will return null, if it can not find an entitiy. In line 351 we would then fall back to the first SEO URL entity. And well, if that one doesn't exist as well, we do not have any SEO URL and therefore would fallback to the default product URL.